### DISCUSSION TITLE
Lost Context propagation SpringKafka / kafka client  - manual instrumentation

### DISCUSSION THREAD

**Discussion by dimitri995** [open-telemetry/opentelemetry-java-instrumentation#7033] [Q&A] [answered]
Hello,

I am using the instrumentation of Spring Kafka but I can't propagate the context in this flow : 
producer1 -> consumer1 (become producer2)-> consumer2.

I would want to have 1 trace ID but I got 2. It seems I lost the context... I got traceparent in headers of the message until the consumer1 became the producer2, it seems to overwrite the existing traceparent.

Have you any idea why it's happening ?

OpenTelemetry is integrated in a lib used by our applications. 

Thanks for your time.

Regards,

Dimitri

![image](https://user-images.githubusercontent.com/66120155/199604783-26082a46-c2b7-4800-b223-fd4373d83759.png)

**Comment by dimitri995** [comment, marked as answer]
I fixed it,
I was using the interceptor class for the listener, see below : 
![image](https://user-images.githubusercontent.com/66120155/199698072-dfde91f2-8484-4748-89f9-e0bede26d171.png)

I am using SpringKafka, so I need to use this interceptor way to propagate context correctly : 

![image](https://user-images.githubusercontent.com/66120155/199698281-4f91e5b6-05c5-40d6-a0e2-f78b440a50c3.png)

Regards,

**Reply by mateuszrzeszutek** [reply]
This might be useful to you: here's an example of using library instrumentation in a (test) spring application to instrument kafka: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-kafka-2.7/library/src/test/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/SpringKafkaNoReceiveTelemetryTest.java
